lib/data/data_sources/local/dao/character_dao.dart

```diff
@@ -0,0 +1,46 @@
+import 'package:drift/drift.dart';
+import 'package:marvelapp_flutter/data/converters/from_table_to_character.dart';
+import 'package:marvelapp_flutter/data/data_sources/local/database/database.dart';
+import 'package:marvelapp_flutter/data/data_sources/local/database_tables/character_table.dart';
+import 'package:marvelapp_flutter/domain/entities/character.dart';
+
+part 'character_dao.g.dart';
+
+@DriftAccessor(tables: [CharacterTable])
+class CharacterDao extends DatabaseAccessor<Database> with _$CharacterDaoMixin {
+  final Database database;
+
+  CharacterDao(this.database) : super(database);
+
+  Future<void> insertData(List<Character> characters) async {
+    var count = database.characterTable.id.count();
+    var query = database.selectOnly(database.characterTable)..addColumns([count]);
+    int numberOfRows = await query.map((row) => row.read(count)).getSingle();
+    if (characters.isNotEmpty && numberOfRows == 0) {
+      for (var element in characters) {
```

Comment:
Please check if batch operations are supported to save List<CharacterTableCompanion> in one go.

Reply:
add batch, and convertor from CharacterToTableData